build_tools/bazel/iree_bitcode_library.bzl

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Rules for compiling with clang to produce bitcode libraries."""
+
+def iree_bitcode_library(
+        name,
+        srcs,
+        hdrs = [],
+        copts = [],
+        defines = [],
+        data = [],
+        out = None,
+        clang_tool = "@llvm-project//clang:clang",
+        link_tool = "@llvm-project//llvm:llvm-link",
+        builtin_headers_dep = "@llvm-project//clang:builtin_headers_gen",
+        builtin_headers_path = "external/llvm-project/clang/staging/include/",
```

Comment:
This is basically relying on an implementation detail of Bazel, which isn't great. It would break any repository aliasing and such. It should be possible to get this via a target with https://bazel.build/rules/lib/Label#workspace_root